hostengine/src/HostEngineCommandLine.cpp

```diff
@@ -146,6 +146,10 @@ std::string ParseBindIp(std::string const &value)
     {
         return ""s;
     }
+    /* If the value is an IPv6 address, then we need to remove [] from it */
+    if (value.size() > 1 && value[0] == '[' && value[value.size() - 1] == ']') {
```

Comment:
Could there be a possible out-of-bounds read/copy here? 
For value.size() - 2 to make sense, I think value.size needs to be at least 3.
```suggestion
    if (value.size() >= 3 && value[0] == '[' && value[value.size() - 1] == ']') {
```